views/js/e2e/runner/tools/tools.spec.js

```diff
@@ -39,46 +39,557 @@ describe('Tools', () => {
+                cy.get('@calc').should('be.visible');
+                cy.get('@calc').find('.calcDisplay').as('display');
+
+                // 2 + 2 => 4
+                cy.get('@calc').find('[data-key="2"]').click();
+                cy.get('@calc').find('[data-key="+"]').click();
+                cy.get('@calc').find('[data-key="2"]').click();
+                cy.get('@calc').find('[data-key="="]').click();
+                cy.get('@display').should('have.value', '4');
+                // clear
+                cy.get('@calc').find('[data-key="C"]').click();
+                cy.get('@display').should('have.value', '0');
+
+                // close it
+                cy.get('@toolBtn').click();
+            });
+
+            it('is dynamic', function() {
+                // TODO: draggable
+                // TODO: resizable
+            });
+        });
+
+        describe('Zoom tool', () => {
+
+            beforeEach(() => {
+                cy.get('.tools-box-list').within(() => {
+                    cy.get('[data-control=zoomOut] a').as('zoomOut');
+                    cy.get('[data-control=zoomIn] a').as('zoomIn');
+                });
+                cy.get('.test-runner-scope .qti-item').as('item');
+            });
+
+            it('loads', function() {
+                cy.get('@zoomOut').should('have.length', 1).and('be.visible');
+                cy.get('@zoomIn').should('have.length', 1).and('be.visible');
+            });
+
+            it('zooms in/out', function() {
+                // zoom out
+                cy.get('@zoomOut').click();
+                cy.get('@item')
+                    .should('have.class', 'transform-scale')
+                    // .and('have.css', 'transform-origin', '0px 0px');
+                    .and('have.attr', 'style').and('contain', 'scaleX(0.9)').and('contain', 'scaleY(0.9)');
+                cy.get('@zoomOut').click();
+                cy.get('@item')
+                    .should('have.class', 'transform-scale')
+                    .and('have.attr', 'style').and('contain', 'scaleX(0.8)').and('contain', 'scaleY(0.8)');
+
+                // reset
+                cy.get('@zoomIn').click();
+                cy.get('@zoomIn').click();
+
+                // zoom in
+                cy.get('@zoomIn').click();
+                cy.get('@item')
+                    .should('have.class', 'transform-scale')
+                    .and('have.attr', 'style').and('contain', 'scaleX(1.1)').and('contain', 'scaleY(1.1)');
+                cy.get('@zoomIn').click();
+                cy.get('@item')
+                    .should('have.class', 'transform-scale')
+                    .and('have.attr', 'style').and('contain', 'scaleX(1.2)').and('contain', 'scaleY(1.2)');
+
+                // beyond the max! (2.0)
+                cy.get('@zoomIn').click().click().click().click().click().click().click().click().click().click();
```

Comment:
Can you use a for loop here, instead of do this 10 times.

Reply:
✅